sycl/doc/EnvironmentVariables.md

```diff
@@ -28,6 +29,53 @@ compiler and runtime.
 
 `(*) Note: Any means this environment variable is effective when set to any non-null value.`
 
+### `ONEAPI_DEVICE_SELECTOR`
+
+With no environment variables set to say otherwise, all platforms and devices presently on the machine are available. The default choice will be one of them, usually preferring a LevelZero GPU device, if available. The `ONEAPI_DEVICE_SELECTOR` can be used to limit that choice of devices, and to expose sub-devices (aka "tiles") as individual devices.  
+
+A value for this environment value consists of a backend name, followed by a colon ( : ), followed by some sort of device specifier (a named device type, a wildcard symbol ( * ) , or an exact numeric device id).  This can optionally be followed by period ( . )  and then a sub-device specifier, which could be either a wilcard symbol ( * ) or an numeric index (using 0 based counting).   Commas can be used to string multiple device specifiers behind any backend.    Moreover, individual entries can be separated by semi-colons. 
+
+
+Possible values of `backend` are:
+- `level_zero`
+- `opencl`
+- `cuda`
+- `hip`
+- `esimd_emulator`
+- `*`
+
+Possible named device types are:
+- `cpu`
+- `gpu`
+- `acc`
+- `*`
+
+To see the exact numeric device id of each of the available devices, simply run `sycl-ls`.
+
+Example: `ONEAPI_DEVICE_SELECTOR=opencl:2.0` This would select the first tile of some device with an id of 2 from OpenCL platform. 
+
+Common Usage Examples:
+
+| Example | Result |
+-----------|---------
+| `ONEAPI_DEVICE_SELECTOR=opencl:*` | All the devices on the OpenCL platform are available |
+| `ONEAPI_DEVICE_SELECTOR=level_zero:gpu` | Only gpu devices on the LevelZero platform will be available.|
+| `ONEAPI_DEVICE_SELECTOR="opencl:gpu;level_zero:gpu"` | GPU devices from both Level Zero and OpenCL will be available. (Quotation marks will likely be needed when using semi-colon separated entries.) |
```

Comment:
I think quotes are not needed in the parsed string itself -- at least not with the Ubuntu 20.04 WSL shell I tried -- although some form of shell escaping is needed.

```
holmesda@holmesda-mobl:~$ export ONEAPI_DEVICE_SELECTOR=level-zero:gpu;opencl:gpu
opencl:gpu: command not found
holmesda@holmesda-mobl:~$ export ONEAPI_DEVICE_SELECTOR='level-zero:gpu;opencl:gpu'
holmesda@holmesda-mobl:~$ echo $ONEAPI_DEVICE_SELECTOR
level-zero:gpu;opencl:gpu
holmesda@holmesda-mobl:~$ export ONEAPI_DEVICE_SELECTOR=level-zero:gpu\;opencl:gpu
holmesda@holmesda-mobl:~$ echo $ONEAPI_DEVICE_SELECTOR
level-zero:gpu;opencl:gpu
```